### DISCUSSION TITLE
Can i make dynamic page with prefix?

### DISCUSSION THREAD

**Discussion by beautyfree** [vercel/next.js#15079] [Help] [answered]
I want to have dynamic pages /projectname and /@username
If i create files [projectAlias].tsx and @[username].tsx it doesnt' work?

What about availability of such an approach?

**Comment by jamesmosier** [comment, marked as answer]
Why does the page name need to be prefixed with `@`? If you have a page `[username].tsx` and then have your url `localhost:3000/@foo` then `router.query` is `@foo`.

**Reply by beautyfree** [reply]
@jamesmosier i want to have url for projects without prefix @ and for users with `@`
My files are called `[projectAlias].tsx` and `@[username].tsx`, you lost @

**Reply by jamesmosier** [reply]
I see. As you've found, there is no built in way to handle 2 dynamic routes in the same folder currently, even with a prefix.

**Reply by david-sling** [reply]
@jamesmosier You can have the filename as [dynamicPage].tsx 

```
const DynamicPage = () => {
  const router = useRouter()
  const { dynamicPage } = router.query
  return dynamicPage[0]==="@" ? <Username /> : <ProjectAlias />
}
